catwalk/steps.py

```diff
@@ -85,7 +86,14 @@ def run(self, metrics: Dict[str, Dict[str, float]], format: str = "text") -> Ite
         if format == "text":
             for task_name, task_metrics in metrics.items():
                 for metric_name, metric_value in task_metrics.items():
-                    yield f"{task_name}\t{metric_name}\t{metric_value}"
+                    # For SQuAD metric a dictionary is returned, so we need to iterate over the keys to get the values
+                    if isinstance(metric_value, dict):
+                        for nested_metric_name, nested_metric_value in metric_value.items():
+                            # For SQuAD metric the return value is a tensor so we need to get the item
+                            nested_metric_value = nested_metric_value.item() if isinstance(nested_metric_value, torch.Tensor) else nested_metric_value
+                            yield f"{task_name}\t{nested_metric_name}\t{nested_metric_value}"
+                    else:
+                        yield f"{task_name}\t{metric_name}\t{metric_value}"
```

Comment:
Can you make this a bit more generic, so we can just support nested metrics in general? Basically, we need a way to flatten the metrics, so that the final table would contain something like `squad    squad_metrics.f1    74.8`.